beluga/docs/README.md

```diff
@@ -1,12 +1,87 @@
-This library provides implementations for Monte Carlo based localization algorithms.
-The existing implementations are:
-
-- MCL: Monte Carlo Localization using a fixed number of particles.
-- AMCL: Adaptive Monte Carlo Localization, the number of particles is defined using the KLD criteria.
-
-The library is extensible, allowing to:
-- Provide different [sensor](@ref SensorModelPage) or [motion](@ref MotionModelPage) models.
-- Provide a way of getting the [estimated pose](@ref StateEstimatorPage) based on the particles.
-- Provide a way of [generating the initial particles](@ref StateGeneratorPage).
-- Provide a way of [sampling](@ref SamplerPage) from the previous particle set.
-- Provide a way to [limiting](@ref LimiterPage) the amount of particles.
+# Beluga
+
+## Introduction
+
+Beluga is a ROS-agnostic C++17 library that provides implementations for Monte Carlo-based localization algorithms widely used in robotics applications.
+Its modularity allows users to compose solutions from reusable modules and to combine them with new ones to configure the MCL algorithm that best suits their needs.
+
+The API extends concepts from the STL, emphasizing support for ranges of particles.
+
+## Components
+
+Explore the library's components organized by concept:
+
+### Sensor models
+
+Probabilistic sensor models that describe the likelihood of obtaining a certain measurement given a specific state of the system. \n
+They satisfy \ref SensorModelPage.
+
+| | |
+|-|-|
+| beluga::BeamSensorModel | Beam sensor model for range finders |
+| beluga::LikelihoodFieldModel | Likelihood field sensor model for range finders |
+| beluga::LandmarkSensorModel | Generic landmark model for discrete detection sensors (both 2D and 3D) |
+| beluga::BearingSensorModel | Generic bearing sensor model for discrete detection sensors (both 2D and 3D) |
+
+### Motion models
+
+Probabilistic motion models that describe how the state of the system evolves over time. \n
+They satisfy \ref MotionModelPage.
+
+| | |
+|-|-|
+| beluga::DifferentialDriveModel | Sampled odometry model for a differential drive |
+| beluga::OmnidirectionalDriveModel | Sampled odometry model for an omnidirectional drive |
+
+### Range views
+
+Useful views for working with ranges of particles.
+They are lazily evaluated range adaptor objects compatible with the Range-v3 library.
+
+| | |
+|-|-|
+| \link views/elements.hpp beluga::views::elements \endlink | Takes a view of tuple-like values and a number N and produces a view of Nth element of each tuple |
```

Comment:
@nahueespinosa I don't think `\link` commands are a thing in Markdown either.

Reply:
They aren't. But this is a Doxygen page, and after the Markdown preprocessor, Doxygen does a second pass to parse special commands and HTML commands.

> When doxygen parses the source code it first extracts the comments blocks, then passes these through the Markdown preprocessor. The output of the Markdown preprocessing consists of text with [special commands](https://www.doxygen.nl/manual/commands.html#cmd_intro) and [HTML commands](https://www.doxygen.nl/manual/htmlcmds.html). A second pass takes the output of the Markdown preprocessor and converts it into the various output formats.

Reply:
I switched to using pure Markdown here: https://github.com/Ekumen-OS/beluga/pull/330/commits/b608def392594d038690e0b41458207f6be052cd

Reply:
For reference, this is how it looks now. There are still open tickets to improve the content, this is just a small improvement to what we had.

![image](https://github.com/Ekumen-OS/beluga/assets/33042669/b44c4f52-9023-4e87-85c7-672d90d216ab)

Reply:
Looks great.